### PR TITLE
Migrates dotnet.exe NuGet sources commands to System.CommandLine

### PR DESCRIPTION
**Pull request by dominoFire**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2094

Regression? Last working version: N/A

## Description

- Migrates remove, disable, enable, update sources commands to System.CommandLine. This code is autogenerated using the templates
- Adds tests to verify both command implementations (System.CommandLine and CommandLineUtils) have the same command output in simple scenarios

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A: Not major product changes.

### CODE REVIEW

**Review by zivkan — APPROVED** [review]

File: test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/Commands/UpdateSourceTests.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            });
+
+            // Act
+            int statusCurrent = currentCli.Execute(enableSourceCmd1);
+            int statusNew = newCli.Invoke(enableSourceCmd2);
+
+            // Assert
+            CommandTestUtils.AssertBothCommandSuccessfulExecution(statusCurrent, statusNew, testLoggerCurrent, testLoggerNew);
+            Assert.NotEqual(initialConfig, File.ReadAllText(file1));
+            Assert.NotEqual(initialConfig, File.ReadAllText(file2));
```
Comment (donnie-msft) [inline test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/Commands/UpdateSourceTests.cs:51]: Based on test title, I was expecting to see the output of both commands compared here. I realize now that this is happening in your Utils. Can `AssertBothCommandSuccessfulExecution` be named to emphasize equality? It sounded to me like it was a simple status code checker.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/Commands.xml [hunk 2]
```diff
@@ -20,6 +20,26 @@
       <SeeAlso Title="sources command (nuget.exe)" Url="/nuget/reference/cli-reference/cli-ref-sources" />
     </Noun>
   </Verb>
+  <Verb Name="disable">
```
Comment (erdembayar) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/Commands.xml:26]: Team review: Please add guide on how to update boilerplate C# code from XML.

Comment (dominoFire) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/Commands.xml:26]: Added a line in this xml file.

File: test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/Commands/UpdateSourceTests.cs (hunk 1 above)
Comment (dominoFire) [reply test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/Commands/UpdateSourceTests.cs:51]: Renamed to AssertEqualCommandOutput

**Comment by dominoFire** [comment]
I had previous approval to merge this PR. So, I'm going to merge it. Any issues, please let me know.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/Commands.xml [hunk 3]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <!-- Copyright (c) .NET Foundation. All rights reserved.
      Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. -->
+<!-- 
+Run CommandParsers.tt, CustomBinders.tt, VerbParsers.tt in this folder to update System.CommandLine command binding code
```
Comment (erdembayar) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/Commands.xml:5]: Sorry, last question, you don't have to commit anything.
How do we run *.tt files? Is it something VS does for us?

Comment (dominoFire) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/Commands.xml:5]: > re-execute the text template via "run custom tool" on VS context menu for .tt file, or via dotnet-t4 global tool.


https://github.com/NuGet/NuGet.Client/blob/4a6b7bbad3fb31eb6d8c5a4242649419dedd0f29/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Verbs.tt#L19

Comment (dominoFire) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/Commands.xml:5]: Visual Studio can run the templates by right clicking the .tt file in Solution Explorer and choosing "Run Custom Tool" or using dotnet-t4 global tool on NuGet.CommandLine.XPlat project.

**Review by erdembayar — APPROVED** [review]

**Pull request merged**